### DISCUSSION TITLE
In Annotations Framework how can I get all parameters passed with a wildcard?

### DISCUSSION THREAD

**Discussion by lorenh** [aws/aws-lambda-dotnet#1683] [Q&A] [answered]
I'm looking at [FromQuery] and see how it works with a well-known parameter, but in cases where there may be a variable number of unknown parameters passed to my HttpApi endpoint, is there a wildcard mechanism, or way to access the raw request information to get the unknown parameters that were passed?

**Comment by ashishdhingra** [comment]
@lorenh Based on my understanding, there isn't a way to specify wildcard mechanism in Annotations Framework. While on high level it might appear to have some kind of relation to ASP.NET framework, using annotations framework leverages source generators, which:
- Generates classes for various methods decorated with `HttpApi` behind the scenes with manually constructed `ServiceCollection` behind the scenes for DI.
- Generates CloudFormation template which maps various endpoints in API Gateway to a Lambda function.
- Uses `APIGatewayHttpApiV2ProxyRequest` and `APIGatewayHttpApiV2ProxyResponse` classes to intercept requests and send response.

Hence, there is no concept of ASP.NET's `Request` object which might have given access to raw HTTP request and query string parameters.

CCing @normj for any other inputs.

Thanks,
Ashish

**Comment by normj** [comment, marked as answer]
As @ashishdhingra mention you can inject the raw request into the function's parameters to get access to the raw query string.

```csharp
[LambdaFunction(Policies = "AWSLambdaBasicExecutionRole", MemorySize = 512, Timeout = 30)]
[RestApi(LambdaHttpMethod.Get, "/{product}")]
public IHttpResult Get(string product, APIGatewayProxyRequest rawRequest, ILambdaContext context)
{
    var builder = new StringBuilder();
    builder.AppendLine("Product: " + product);
    if (rawRequest.QueryStringParameters != null)
    {
        builder.AppendLine("Query String:");
        foreach (var query in rawRequest.QueryStringParameters)
        {
            builder.AppendLine($"\t{query.Key}: {query.Value}");
        }
    }

    context.Logger.LogInformation(builder.ToString());

    return HttpResults.Ok(builder.ToString())
                    .AddHeader("Content-Type", "text/plain");
}
```
